engine/platform/sdl/events.c

```diff
@@ -356,6 +356,38 @@ static void SDLash_ActiveEvent( int gain )
 	}
 }
 
+#if SDL_VERSION_ATLEAST( 2, 0, 0 )
+static size_t num_open_game_controllers = 0;
+
+static void SDLash_GameController_Add( int index )
+{
+	extern convar_t *joy_enable; // private to input system
+	SDL_GameController *controller;
+	if ( !joy_enable->value ) return;
+	controller = SDL_GameControllerOpen( index );
+	if (controller== NULL) {
+		Con_Reportf( "Failed to open SDL GameController %d: %s\n", index, SDL_GetError( ) );
+		SDL_ClearError( );
+		return;
+	}
+#if SDL_VERSION_ATLEAST( 2, 0, 6 )
+	Con_Reportf( "Added controller: %s (%i:%i:%i)\n",
+		SDL_GameControllerName( controller ),
+		SDL_GameControllerGetVendor( controller ),
+		SDL_GameControllerGetProduct( controller ),
+		SDL_GameControllerGetProductVersion( controller ));
+#endif // SDL_VERSION_ATLEAST( 2, 0, 6 )
+	if ( ++num_open_game_controllers == 1 ) Joy_AddEvent( );
```

Comment:
```suggestion
	if( ++num_open_game_controllers == 1 )
		Joy_AddEvent( );
```

Wait, why 1 though? What if we connect more than 1 controller? I know, we don't use it at this moment and there is no mod that uses multiple gamepads(Decay could be one of them but there is no source code of Decay :)), but even if it's unused, we still need to notify engine about them.

Also, I don't really like the `++x == y` construction. Maybe put increment somewhere else, so check will be much more obvious?

Reply:
We only want to call `Joy_Add/RemoveEvent` once because these methods set the `joy_found` variable.

If we called it every time, the following sequence would break `joy_found`

* Add gamepad 1 -- joy_found is 1
* Add gamepad 2 -- joy_found is 1
* Unplug gamepad 2 -- joy_found is 0 -- wrong

Instead, we only call `Add` when activating the first joystick and `Remove` when removing the last joystick.

TLDR: The `Joy_Add/RemoveEvent` code assumes only a single joysticks

Reply:
I've also added a comment in `SDLash_GameController_Remove` explaining the check.
The check in `Add` is not strictly necessary with the current `Joy_AddEvent` implementation.

Reply:
>The Joy_Add/RemoveEvent contract seems to assume only a single joystick

I wanted to get all events from the operating system, maybe to do something with them in the future. But you're right, we don't do anything with this and I doubt we will.

Marking as resolved then.